pypit/arutils.py

```diff
@@ -1076,6 +1076,61 @@ def subsample(frame):
     indices = coordinates.astype('i')
     return frame[tuple(indices)]
 
+def trace_gweight(fimage, xcen, ycen, sigma, invvar=None):
+    """ Port of SDSS trace_gweight algorithm
+    Parameters
+    ----------
+    fimage
+    xcen
+    ycen
+    sigma
+    invvar
+
+    Returns
+    -------
+    xnew
+    xerr
+
+    """
+    # import
+    from scipy.special import erf
+    # Setup
+    nx = fimage.shape[1]
+    xnew = xcen.copy()
+    xerr = 0.0 * xnew # Allocate memory
```

Comment:
Actually, given that below (line 1127) we set all values of xerr to be masked, I suggest we initialize the mask here. I also propose that we add a keyword argument to this function `maskval=-999999.9`. I suggest that we use this value as a mask for now, since all of the cython code assumes this value, and it's probably best to stay consistent (at least for now). So this line should probably read: `xerr = maskval*np.ones_like(xnew)`, and then delete line 1127.